packages/lib/src/vis/utils.ts

```diff
@@ -350,19 +350,39 @@ export function toArray(arr: NumArray): number[] {
   return isTypedArray(arr) ? [...arr] : arr;
 }
 
-export function worldToCamera(camera: Camera, worldPt: Vector3) {
-  return worldPt.clone().project(camera);
+export function worldToHtml(
+  camera: Camera,
+  context: VisCanvasContextValue,
+  worldPt: Vector3
+) {
+  const cameraPt = worldPt.clone().project(camera);
+  return context.cameraToHtml(cameraPt);
+}
+
+export function htmlToWorld(
+  camera: Camera,
+  context: VisCanvasContextValue,
+  htmlPt: Vector2
+) {
+  return context.htmlToCamera(htmlPt).unproject(camera);
 }
 
 export function dataToHtml(
   camera: Camera,
   context: VisCanvasContextValue,
   dataPt: Vector2
 ): Vector2 {
-  const { dataToWorld, cameraToHtml } = context;
-  const worldPt = dataToWorld(dataPt);
-  const cameraPt = worldToCamera(camera, worldPt);
-  return cameraToHtml(cameraPt);
+  const worldPt = context.dataToWorld(dataPt);
+  return worldToHtml(camera, context, worldPt);
+}
+
+export function htmlToData(
+  camera: Camera,
+  context: VisCanvasContextValue,
+  htmlPt: Vector2
+): Vector2 {
+  const worldPt = htmlToWorld(camera, context, htmlPt);
+  return context.worldToData(worldPt);
```

Comment:
Or even better `dataToHtml(htmlToData(vec))` to test the full chain of transformations

Reply:
I just tried and it does indeed give back roughly the same vector (with some extra tiny decimals as expected).